File: java/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> hash = new HashMap();
        for (int i=0;i<nums.length;i++) {
            int v = nums[i];
            if (hash.containsKey(target-v)) {
                return new int[]{hash.get(target-v), i};
            }
            hash.put(v, i);
        }
        return null;
    }
}

File: cpp/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> hash;
        for (int i=0;i<nums.size();i++) {
            int v = nums[i];
            if (hash.find(target-v) != hash.end()) {
                return vector<int>{hash[target-v], i};
            }
            hash[v] = i;
        }
        return vector<int>{};
    }
};
